### PR TITLE
Add javascript implementation

### PR DESCRIPTION
**Pull request by romedu**

### CODE REVIEW

**Comment by Fleshgrinder** [comment]
This has conflicts by now because other solutions where provided. @romedu would it be possible for you to split this PR into smaller ones?

File: problems/alphanumeric-string-sort/alphanumeric-string-sort.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      else if(character.toUpperCase() === character) uppercaseLetters.push(character);
+      else lowercaseLetters.push(character);
+   })
+
+   return lowercaseLetters.sort().concat(uppercaseLetters.sort()).concat(evenDigits.sort()).concat(oddDigits.sort()).join("");
+}
+
+console.log(alphanumericStringSort("Sorting0123456789"));            // ginortS0246813579
+console.log(alphanumericStringSort("foobar1237348421"));             // abfoor2244811337
+console.log(alphanumericStringSort("789765445whjdbjwhwfbs977865"));  // bbdfhhjjswww446688555777799
```
Comment (Fleshgrinder) [inline problems/alphanumeric-string-sort/alphanumeric-string-sort.js]: We have a simpler solution for this in #135.

**Comment by romedu** [comment]
> This has conflicts by now because other solutions where provided. @romedu would it be possible for you to split this PR into smaller ones?

sure

File: problems/alphanumeric-string-sort/alphanumeric-string-sort.js (hunk 1 above)
Comment (romedu) [reply problems/alphanumeric-string-sort/alphanumeric-string-sort.js]: got it

**Pull request closed without merging**